geode-core/src/main/java/org/apache/geode/internal/cache/EntryEventImpl.java

```diff
@@ -2871,12 +2913,14 @@ public void release() {
    * Return true if this EntryEvent may have off-heap references.
    */
   private boolean mayHaveOffHeapReferences() {
-    InternalRegion lr = getRegion();
-    if (lr != null) {
-      return lr.getOffHeap();
-    }
+    /*
+     * InternalRegion lr = getRegion();
+     * if (lr != null) {
+     * return lr.getOffHeap();
+     * }
+     */
     // if region field is null it is possible that we have off-heap values
-    return true;
+    return this.useOffHeap;
```

Comment:
Since you have tests that now just pass in "false" to the constructor about off-heap I think you can restore the old implementation of this method and then add a check at the top for off-heap being available (offHeapLock != null) and if not return false. Then do the old impl.